File: file_sharing_mobile/lib/controllers/converter.dart
class MiConverter {

  static List<int> intToBytes(int value) {
    List<int> bytes = [];
    for (int i = 3; i >= 0; i--) {
      bytes.add((value >> (i * 8)) & 0xFF);
    }
    return bytes;
  }

  static int bytesToInt(List<int> bytes) {
    if (bytes.length != 4) {
      throw ArgumentError('El arreglo de bytes debe tener una longitud de 4');
    }
    int value = 0;
    for (int i = 0; i < 4; i++) {
      value = (value << 8) | bytes[i];
    }
    return value;
  }
}

File: file-sharing-desktop/app/src/main/java/com/example/file/sharing/controllers/Converter.java
package com.example.file.sharing.controllers;

import java.nio.ByteBuffer;

public class Converter {

    public static byte[] intToBytes(int valor) {
        return ByteBuffer.allocate(4).putInt(valor).array();
    }

    public static int bytesToInt(byte[] bytes) {
        if (bytes.length != 4) {
            throw new IllegalArgumentException("El arreglo debe tener 4 bytes");
        }
        return ByteBuffer.wrap(bytes).getInt();
    }
}
